HHS/Head-Start-TTADP
# [TTAHUB-1740] Allow groups to be marked public

## Description of change
Adds the ability to make groups public. A public group is viewable by anyone with permissions in the same region as the grants in the group.

## How to test
Confirm the following:
- You can create a public group (a checkbox has been added to the form, checked by default, indicating that a group is private)
- By impersonating or switching to a different user, you can view that group under the appropriate section, both as a filter option and through the "view" page accessibile via the groups table
- You cannot reuse a group name (case insensitive, checking for leading and trailing whitespace)
- You can edit and delete all your groups as normal 

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1740


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
